src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -150,18 +282,128 @@ public class PipelineCacheServer
             return pipelineCacheClient;
         }
 
-        private Task DownloadPipelineCacheAsync(
+        private async Task DownloadPipelineCacheAsync(
+            AgentTaskPluginExecutionContext context,
             DedupManifestArtifactClient dedupManifestClient,
             DedupIdentifier manifestId,
             string targetDirectory,
+            bool isTar,
+            string contentFormat,
             CancellationToken cancellationToken)
         {
-            DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
-                manifestId,
-                targetDirectory,
-                proxyUri: null,
-                minimatchPatterns: null);
-            return dedupManifestClient.DownloadAsync(options, cancellationToken);
+
+            // Throw for bunch of invalid combinations of istar and contentFormat.
+            if (!isTar)
+            {
+                DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
+                    manifestId,
+                    targetDirectory,
+                    proxyUri: null,
+                    minimatchPatterns: null);
+
+                await dedupManifestClient.DownloadAsync(options, cancellationToken);
+            }
+            else
+            {
+                DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
+                    manifestId,
+                    targetDirectory,
+                    proxyUri: null,
+                    minimatchPatterns: null);
+
+                options.IsTar = true;
+
+                var processTcs = new TaskCompletionSource<int>();
+
+                using (var cancelSource = new CancellationTokenSource())
+                using (var linkedSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, cancelSource.Token))
+                using (var process = new Process())
+                {
+                    process.StartInfo.FileName = isWindows ? "7z" : "tar";
```

Comment:
I am unable to start the process using just the "7z", even after adding the PATH variable, any thoughts ?